src/App/Fossa/FossaAPIV1.hs

```diff
@@ -430,6 +476,26 @@ archiveUpload signedArcURI arcFile = fossaReq $ do
   where
     uploadArchiveRequest url options = reqCb PUT url (ReqBodyFile arcFile) lbsResponse options (pure . requestEncoder)
 
+---------- The license scan result upload function uploads the JSON license result directly to the signed URL it is provided.
+
+licenseScanResultUpload ::
+  (Has (Lift IO) sig m, Has Diagnostics sig m) =>
+  SignedURL ->
+  LicenseSourceUnit ->
+  m LbsResponse
+licenseScanResultUpload signedArcURI licenseScanResult = fossaReq $ do
+  let arcURL = URI.mkURI $ signedURL signedArcURI
+
+  uri <- fromMaybeText ("Invalid URL: " <> signedURL signedArcURI) arcURL
+  validatedURI <- fromMaybeText ("Invalid URI: " <> toText (show uri)) (useURI uri)
+
+  context ("Uploading license scan result to " <> signedURL signedArcURI) $ case validatedURI of
+    Left (url, options) -> uploadArchiveRequest url options
```

Comment:
[Prefer] I know in the codebase this is previously used but lets name Left/Right with scheme for readability.
```haskell
    Left (httpUrl, httpOption) -> uploadArchiveRequest httpUrl httpOption
    Right (httpsUrl, httpsOptions) -> uploadArchiveRequest httpsUrl httpsOptions
```